receiver/gitproviderreceiver/internal/scraper/githubscraper/github_scraper.go

```diff
@@ -108,12 +108,63 @@ func (ghs *githubScraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
 		go func() {
 			defer wg.Done()
 
-			count, err := ghs.getBranches(ctx, genClient, name, trunk)
+			branches, count, err := ghs.getBranches(ctx, genClient, name, trunk)
 			if err != nil {
 				ghs.logger.Sugar().Errorf("error getting branch count for repo %s", zap.Error(err), repo.Name)
 			}
 			ghs.mb.RecordGitRepositoryBranchCountDataPoint(now, int64(count), name)
 
+			// Iterate through the branches populating the Branch focused
+			// metrics
+			for _, branch := range branches {
+
+				// For each branch we will check to make sure it's not the
+				// default branch or a branch with no changes to it. `BehindBy`
+				// in this conditional actually means `AheadOf` the default
+				// branch. This is a byproduct of the way the GraphQL queries
+				// are made (performance reasons) and the information that is
+				// returned.This is done for three main reasons.
+				// 1. The default branch will always be a long-lived branch and
+				//    may end up with more commits than can be possibly queried
+				//    at a given time.
+				// 2. The default is the trunk of which all changes should go
+				//    into. The intent of these metrics is to provide useful
+				//    signals helping identify cognitive overhead and
+				//    bottlenecks.
+				// 3. GitHub does not provide any means to determine when a
+				//    branch was actually created. Git the tool however does
+				//    provide a created time for each ref off the trunk. GitHub
+				//    does not expose this data via their API's and thus we
+				//    have to calculate age based on commits added to the
+				//    branch.
+				if branch.Name == branch.Repository.DefaultBranchRef.Name || branch.Compare.BehindBy == 0 {
+					continue
+				}
+
+				// As mentioned in the comment preceding this, `BehindBy`
+				// actually means `AheadBy` the default branch and `AheadBy`
+				// means `BehindBy` the default branch. This is a byproduct of
+				// the GraphQL queries being made.
+				ghs.mb.RecordGitRepositoryBranchCommitAheadbyCountDataPoint(now, int64(branch.Compare.BehindBy), branch.Repository.Name, branch.Name)
+				ghs.mb.RecordGitRepositoryBranchCommitBehindbyCountDataPoint(now, int64(branch.Compare.AheadBy), branch.Repository.Name, branch.Name)
+
+				var additions int
+				var deletions int
+				var age int64
+
+				// TODO: rename getCommitInfo
```

Comment:
Reminder for me to remove this comment after turning `getCommitInfo` into `evalCommits`